### ISSUE TITLE
Test report file (txt) not created

### ISSUE THREAD

**Issue by doronkatz** [realm/SwiftLint#2977]
### New Issue Checklist

- [x ] Updated SwiftLint to the latest version
- [x ] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

When creating a CI in Bitbucket pipelines using SwiftLint, I get the following warning in bitbucket:
```bash
Searching for test report files in directories named [test-results, failsafe-reports, test-reports, surefire-reports] down to a depth of 4
Finished scanning for test reports. Found 0 test report files.
Merged test suites, total number tests is 0, with 0 failures and 0 errors.

```
In other words it doesn't generate a report file. 

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
+ swiftlint autocorrect
Loading configuration from '.swiftlint.yml'
Correcting Swift files at paths 
Correcting 'Package.swift' (1/11)
Correcting 'Popular.swift' (2/11)
Correcting 'Genre.swift' (3/11)
Correcting 'Paginator.swift' (4/11)
Correcting 'ContentItem.swift' (5/11)
Correcting 'APIService.swift' (6/11)
Correcting 'Token.swift' (7/11)
Correcting 'PaginatedResponse.swift' (8/11)
Correcting 'XCTestManifests.swift' (9/11)
Correcting 'SwiftRecommendationsTests.swift' (10/11)
Correcting 'LinuxMain.swift' (11/11)
Done correcting 11 files!
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)?
* Installation method used (Homebrew, CocoaPods, building from source, etc)?
* Paste your configuration file:

```yml
excluded:
  - .build/checkouts
line_length: 250
disabled_rules:
  - unused_enumerated
excluded:
  - .build/checkouts
line_length: 250
disabled_rules:
  - unused_enumerated
reporter:
  - "json"
```

**Comment by marcelofabri** [comment]
SwiftLint never creates any files. If you want, you can redirect `stdout` to a file with `> file.txt`
